lightningd/options.c

```diff
@@ -266,6 +268,9 @@ static void config_register_opts(struct lightningd *ld)
 			       ld,
 			       "Select the network parameters (bitcoin, testnet,"
 			       " regtest, or litecoin)");
+	opt_register_arg("--deprecated-apis", opt_set_bool_arg, opt_show_bool,
```

Comment:
Option name seems to imply it would *enable* deprecated api.

Reply:
No, wait, the option name is correct.  The description is misleading as it starts with "Disable...", implying that adding `--deprecated-apis` disables the deprecated api, when it really means that `--deprecated-apis=false` that disables the deprecated API.

Reply:
Echoing @ZmnSCPxj concern about how to interpret `--deprecated-apis`.

Perhaps `--no-deprecated-apis` would be less confusing. That would be in line with `--no-reconnect` (`Disable automatic reconnect attempts`).

Reply:
My thinking is that when you upgrade lightningd, stuff should *NOT* break.  But if you're the author of something based on lightningd, you should *TEST* with this option on to see if you're going to break soon.

I'll defer to @shesek here though....

Reply:
And yes, description is a leftover from when it was called '--no-deprecated'.  Will fix!

The problem with *that* is that you can't override it unless we add --deprecated as well (you might want to override a config option on the cmdline, or, for testing, add --deprecated-apis=true after --deprecated-apis=false.

Reply:
@rustyrussell Good point about overriding.

Perhaps using `--allow-deprecated-apis=true` and `--allow-deprecated-apis=false` would make the meaning obvious?

Reply:
Agreed!  Made that change now, will merge once CI happy.